tests/parser/test_jsonschema.py

```diff
@@ -464,3 +467,72 @@ def test_no_additional_imports():
         source='',
     )
     assert len(new_parser.imports) == 0
+
+
+@pytest.mark.parametrize(
+    'source_obj,generated_classes',
+    [
+        (
+            {
+                '$id': 'https://example.com/person.schema.json',
+                '$schema': 'http://json-schema.org/draft-07/schema#',
+                'title': 'Person',
+                'type': 'object',
+                'properties': {
+                    'firstName': {
+                        'type': 'string',
+                        'description': "The person's first name.",
+                        'alt_type': 'integer',
+                    },
+                    'lastName': {
+                        'type': 'string',
+                        'description': "The person's last name.",
+                        'alt_type': 'integer',
+                    },
+                    'age': {
+                        'description': 'Age in years which must be equal to or greater than zero.',
+                        'type': 'integer',
+                        'minimum': 0,
+                        'alt_type': 'number',
+                    },
+                    'real_age': {
+                        'description': 'Age in years which must be equal to or greater than zero.',
+                        'type': 'integer',
+                        'minimum': 0,
+                    },
+                },
+            },
+            """class Person(BaseModel):
+    firstName: Optional[int] = None
+    lastName: Optional[int] = None
+    age: Optional[confloat(ge=0.0)] = None
+    real_age: Optional[conint(ge=0)] = None""",
+        ),
+    ],
+)
+@pytest.mark.skipif(
+    pydantic.VERSION < '2.0.0', reason='Require Pydantic version 2.0.0 or later '
+)
+def test_json_schema_parser_extension(source_obj, generated_classes):
+    """
+    Contrived example to extend the JsonSchemaParser to support an alt_type, which
+    replaces the type if present.
+    """
+
+    class AltJsonSchemaObject(JsonSchemaObject):
+        properties: Optional[Dict[str, Union[AltJsonSchemaObject, bool]]] = None
+        alt_type: Optional[str] = None
+
+        def model_post_init(self, context: Any) -> None:
```

Comment:
> I think the issue is my use of `model_post_init`. That does not appear to be documented in Pydantic v1. Maybe switching it to using `__post_init__` might work? I'll try to investigate this in the morning.

This was not entirely correct, I can fix the failure in Pydantic 1.7+, via 
```suggestion
        def __init__(self, *args, **kwargs):
            super().__init__(*args, **kwargs)
```
However, I run into pydantic/pydantic#1985 on versions older than that. Adjusting the skip down to `< 1. 7` does not prevent the failure in this case as it occurs during test collection not test running.

So its probably best to leave the test as is, and keep the skip.

Reply:
@WilliamJamieson 
Thank you for your research.
Perhaps this is not a problem with the code generator, but with how Pydantic is used.
It should be the responsibility of the creator of the custom schema model.